### PR TITLE
Format date if a custom format has been passed

### LINKED ISSUE #69: Doesn't respect custom format supplied in `data-o-date-format`
According to the docs, you are able to pass a custom format to the above attribute:

> Any other string using widespread conventions for time/date placeholders, which will be replaced with values extracted from the date provided. See './main.js' for an up to date list of supported formats. To avoid e.g. the 'mm' in 'common' being replaced with the month prefix with a double backslash 'co\mmon' i.e. In most cases custom date formats should not be used, in favour of the standard FT date and datetime formats'

However, in this section: https://github.com/Financial-Times/o-date/blob/master/main.js#L96-L104 the formatting is only done if it is in one of the supported formats. 

Should this be the case? Are we more strictly enforcing this statement: 

> In most cases custom date formats should not be used, in favour of the standard FT date and datetime formats'

In which case, we should update the docs...

Or should we fix this to format the date when a custom format has been provided?

### PR DESCRIPTION
**Pull request by onishiweb**
Closes #69

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -99,6 +99,8 @@ ODate.prototype.update = function() {
 		dateString = ODate.format(date, 'date');
 	} else if (format === 'time-ago-limit-4-hours'){
 		dateString = ODate.timeAgo(date, { limit: 4*inSeconds.hour });
+	} else if (format !== '') {
```
Comment (JakeChampion) [inline main.js]: This conditional would mean we fall into this branch of code if `data-o-date-format` is not set, which is what the else statement is meant to catch.

If `data-o-date-format` is not set, it returns `null`, perhaps this conditional should check if `format !== null`, meaning it has been set to something.

Comment (JakeChampion) [inline main.js]: What happens if `data-o-date-format` is set to something non-sensical such as "US"?

Comment (onishiweb) [inline main.js]: Ok, I'm unsure how to go about fixing it. There's a lot of different formatting options (https://github.com/Financial-Times/o-date/blob/master/main.js#L17-L35) and to do a check against all of those different combinations would be crazy? Any thoughts?

Comment (JakeChampion) [inline main.js]: Agree with @roland-vachter on this, I think all we should do here is check that it is not `null` and pass it through to `ODate.format`.

**Comment by JakeChampion** [comment]
All looking good, would it be possible to add a test for this new feature?

**Comment by roland-vachter** [comment]
@JakeChampion actually, this shouldn't be a new feature. It is documented, just that it didn't work. Not sure about the tests, if there is one or not. If it was not caught earlier, there probably isn't one.

**Comment by onishiweb** [comment]
Currently there are no test for this feature, there are actually no tests with any mocks for the declarative method of using `o-date` I can try to add them today but I'm at a conference so it might take me a while or will have to be finished tomorrow.

**Comment by roland-vachter** [comment]
This is blocking a story, so if it works, let's release it, then create the tests later.

**Comment by JakeChampion** [comment]
@roland-vachter Have you confirmed it works?

**Comment by roland-vachter** [comment]
Just tested it, and it works.

**Pull request merged**